### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by AleksandrPok**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,68 @@
...
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
+        <maven.checkstyle.plugin.configLocation>
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (less08) [inline pom.xml]: We don't really need this dependency, since we did't write any tests

File: src/main/java/injectorpractice/model/Car.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.List;
+
+public class Car {
+    private long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(long id, String model, Manufacturer manufacturer, List<Driver> drivers) {
```
Comment (less08) [inline src/main/java/injectorpractice/model/Car.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/03_dao_service_checklist.html
last point

File: src/main/java/injectorpractice/model/Car.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package injectorpractice.model;
+
+import java.util.List;
+
+public class Car {
+    private long id;
```
Comment (less08) [inline src/main/java/injectorpractice/model/Car.java]: Use Wrapper

File: src/main/java/injectorpractice/model/Driver.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package injectorpractice.model;
+
+public class Driver {
+    private long id;
```
Comment (less08) [inline src/main/java/injectorpractice/model/Driver.java]: Use Wrapper

File: src/main/java/injectorpractice/model/Manufacturer.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package injectorpractice.model;
+
+public class Manufacturer {
+    private long id;
```
Comment (less08) [inline src/main/java/injectorpractice/model/Manufacturer.java]: Wrapper

**Review by less08 — CHANGES_REQUESTED** [review]

**Review by less08 — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -8,5 +8,60 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 </project>
```
Comment (Sofasmile) [inline pom.xml]: let's add empty line at the end of the file

File: src/main/java/injectorpractice/dao/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers.stream().filter(m -> m.getId().equals(id)).findFirst();
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/dao/ManufacturerDaoImpl.java]:
```suggestion
        return Storage.manufacturers.stream()
                .filter(m -> m.getId().equals(id))
                .findFirst();
```

File: src/main/java/injectorpractice/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    public Manufacturer update(Manufacturer manufacturer) {
+        IntStream.range(0, Storage.manufacturers.size())
+                .filter(i -> (Storage.manufacturers.get(i).getId().equals(manufacturer.getId())))
+                .findFirst().ifPresent(i -> Storage.manufacturers.set(i, manufacturer));
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/dao/ManufacturerDaoImpl.java]:
```suggestion
                .findFirst()
                .ifPresent(i -> Storage.manufacturers.set(i, manufacturer));
```

File: src/main/java/injectorpractice/model/Car.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(Long id, String model, Manufacturer manufacturer) {
+        this.id = id;
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = new ArrayList<>();
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/model/Car.java]:
```suggestion
        drivers = new ArrayList<>();
```

File: src/main/java/injectorpractice/model/Car.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(Long id, String model, Manufacturer manufacturer) {
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/model/Car.java]: should pass id via constructor?

File: src/main/java/injectorpractice/model/Car.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        this.id = id;
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = new ArrayList<>();
+    }
+
+    public long getId() {
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/model/Car.java]:
```suggestion
    public Long getId() {
```

File: src/main/java/injectorpractice/model/Driver.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
+package injectorpractice.model;
+
+public class Driver {
+    private Long id;
+    private String name;
+    private String licenceNumber;
+
+    public Driver(Long id, String name, String licenceNumber) {
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/model/Driver.java]: the same

File: src/main/java/injectorpractice/service/ManufacturerServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+import injectorpractice.dao.ManufacturerDao;
+import injectorpractice.lib.Inject;
+import injectorpractice.lib.Service;
+import injectorpractice.model.Manufacturer;
+import java.util.List;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    ManufacturerDao manufacturerDao;
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/service/ManufacturerServiceImpl.java]: how about access modifier?

File: src/main/java/injectorpractice/service/ManufacturerServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        manufacturerDao.create(manufacturer);
+        return manufacturer;
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/service/ManufacturerServiceImpl.java]:
```suggestion
        return manufacturerDao.create(manufacturer);
```

File: src/main/java/injectorpractice/service/ManufacturerServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return manufacturerDao.getAll();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        manufacturerDao.update(manufacturer);
+        return manufacturer;
```
Comment (Sofasmile) [inline src/main/java/injectorpractice/service/ManufacturerServiceImpl.java]:
```suggestion
        return manufacturerDao.update(manufacturer);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**